# The E-Commerce Fraud Detection (Chi-Square Goodness-of-Fit Test)
### Objective:

Analyze 25,000 real-world e-commerce transactions across three status categories (Legitimate, Suspicious, Confirmed Fraud) to evaluate whether actual production telemetry deviates significantly from theoretical baseline proportions using the Chi-Square Goodness-of-Fit test.

>Test Used: Chi-Square Goodness of Fit Test

>Performed by : Raghav Katyal on 01 October 2026

In [70]:
import numpy as np
from scipy import stats

In [71]:
#Baseline based on historical records
# 65% transactions fall under Legitimate
# 30% fall under Suspicious
# 5% fall under Confirmed Fraud
baseline = np.array([0.65,0.30,0.05])

In [72]:
#Sample Size = 25000
size=25000
expected_transactions = baseline * size

In [73]:
#Expected Frequency
print(expected_transactions)

[16250.  7500.  1250.]


In [74]:

observed_ratio = np.array([0.63,0.34,0.03])
#Observed Frequency

observed_transactions = observed_ratio * size
print(observed_transactions)

[15750.  8500.   750.]


In [75]:
#Running the Chi-Square Goodness of Fit Test
chi_stat,p_value = stats.chisquare(f_obs=observed_transactions,f_exp=expected_transactions)

In [76]:
print(f"The Chi-Statistic is : {chi_stat:.3f}")
print(f"The P-value is : {p_value:.4f}")

The Chi-Statistic is : 348.718
The P-value is : 0.0000


>Observation : A Chi-Statistic of 348.718 and p-value of 0.000 clearly shows strong evidence against Goodness of Fit ; the data fails the test. Hence, it is observed that the output did not match to the baseline.